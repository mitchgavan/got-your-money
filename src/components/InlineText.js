import styled from 'styled-components'
import { space, width, fontSize, color } from 'styled-system'

export default styled.span`
  ${space}
  ${width}
  ${fontSize}
  ${color}
`
