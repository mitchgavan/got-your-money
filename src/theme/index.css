body {
  font-family: 'Roboto', sans-serif;
  background-color: #f3f4f5;
}
