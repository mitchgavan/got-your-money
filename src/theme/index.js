export default {
  breakpoints: [
    32, 48, 64
  ],
  space: [
    0, 6, 12, 18, 24
  ],
  fontSizes: [
    12, 16, 18, 24, 36, 72
  ],
  colors: {
    black: '#111',
    blue: '#07c',
    green: '#3adb76'
  }
}
