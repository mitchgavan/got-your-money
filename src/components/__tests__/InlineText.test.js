import React from 'react'
import InlineText from '../InlineText'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import theme from '../../theme'

let component

beforeEach(() => {
  component = shallow(
    <InlineText theme={theme} color="black" p={2} fontSize={3} width={1 / 2} />
  )
})

it('renders correctly', () => {
  expect(toJson(component)).toMatchSnapshot()
})

test('it applies styles according to passed props', () => {
  expect(component).toHaveStyleRule('color', theme.colors.black)
  expect(component).toHaveStyleRule('width', '50%')
  expect(component).toHaveStyleRule('padding', theme.space[2] + 'px')
  expect(component).toHaveStyleRule('font-size', theme.fontSizes[3] + 'px')
})
