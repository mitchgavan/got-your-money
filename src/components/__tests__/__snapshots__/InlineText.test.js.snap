// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  padding: 12px;
  width: 50%;
  font-size: 18px;
  color: #111;
}

<span
  className="c0"
  color="black"
  fontSize={3}
  width={0.5}
/>
`;
